src/sentry/similarity.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import absolute_import
+
+import math
+import random
+
+
+def scale_to_total(value):
+    """\
+    Convert a mapping of distinct quantities to a mapping of proportions of the
+    total quantity.
+    """
+    total = float(sum(value.values()))
+    return {k: (v / total) for k, v in value.items()}
+
+
+def get_distance(target, other):
+    """\
+    Calculate the N-dimensional Euclidian between two mappings.
+
+    The mappings are used to represent sparse arrays -- if a key is not present
+    in both mappings, it's assumed to be 0 in the mapping where it is missing.
+    """
+    return math.sqrt(
+        sum(
+            (target.get(k, 0) - other.get(k, 0)) ** 2
+            for k in set(target) | set(other)
+        )
+    )
+
+
+def get_similarity(target, other):
```

Comment:
Realistically if we branch out to other strategies of distance (context matching / tf-idf / whatever) this may be independent of the `MinHashIndex`, so it might be worth getting a blurb in about what's actually happening with the constants below, or maybe just something like:

> Given a pair of previously recorded groups, calculate the aggregate similarity between them by normalizing the per-band distance to be the percent difference.

But dont @ me.